### ISSUE TITLE
1.3.0 can't read messages other than inbox

### ISSUE THREAD

**Issue by laurentiu2** [nextcloud/mail#2917]
### Expected behavior
When selecting a message from 'Sent' or 'Trash' one would expect to see the content of that message.

### Actual behavior
Only messages from inbox are shown.  Anything else is 'loading' and errors are piling up.
OCA\Mail\Exception\ServiceException: Sync failed:

    /var/www/nextcloud/apps/mail/lib/Service/Sync/SyncService.php - line 107:

    OCA\Mail\Service\Sync\ImapToDbSynchronizer->sync(OCA\Mail\Account {}, OCA\Mail\Db\Mailbox { id: 1}, 42, [], true)

    /var/www/nextcloud/apps/mail/lib/Controller/FoldersController.php - line 124:

    OCA\Mail\Service\Sync\SyncService->syncMailbox(OCA\Mail\Account {}, "INBOX.Trash", 42, [], false, null)

    /var/www/nextcloud/lib/private/AppFramework/Http/Dispatcher.php - line 170:

    OCA\Mail\Controller\FoldersController->sync(1, "SU5CT1guVHJhc2g=", [], true, null)

    /var/www/nextcloud/lib/private/AppFramework/Http/Dispatcher.php - line 99:

    OC\AppFramework\Http\Dispatcher->executeController(OCA\Mail\Con ... {}, "sync")

    /var/www/nextcloud/lib/private/AppFramework/App.php - line 125:

    OC\AppFramework\Http\Dispatcher->dispatch(OCA\Mail\Con ... {}, "sync")

    /var/www/nextcloud/lib/private/AppFramework/Routing/RouteActionHandler.php - line 47:

    OC\AppFramework\App::main("OCA\\Mail\\ ... r", "sync", OC\AppFramew ... {}, { accountId: ... "})

    <<closure>>

    OC\AppFramework\Routing\RouteActionHandler->__invoke({ accountId: ... "})

    /var/www/nextcloud/lib/private/Route/Router.php - line 299:

    call_user_func(OC\AppFramew ... {}, { accountId: ... "})

    /var/www/nextcloud/lib/base.php - line 1008:

    OC\Route\Router->match("/apps/mail/ ... c")

    /var/www/nextcloud/index.php - line 38:

    OC::handleRequest()

Caused by OCA\Mail\Exception\IncompleteSyncException: 

    /var/www/nextcloud/apps/mail/lib/Service/Sync/ImapToDbSynchronizer.php - line 146:

    OCA\Mail\Service\Sync\ImapToDbSynchronizer->runInitialSync(OCA\Mail\Account {}, OCA\Mail\Db\Mailbox { id: 1})

    /var/www/nextcloud/apps/mail/lib/Service/Sync/SyncService.php - line 107:

    OCA\Mail\Service\Sync\ImapToDbSynchronizer->sync(OCA\Mail\Account {}, OCA\Mail\Db\Mailbox { id: 1}, 42, [], true)

    /var/www/nextcloud/apps/mail/lib/Controller/FoldersController.php - line 124:

    OCA\Mail\Service\Sync\SyncService->syncMailbox(OCA\Mail\Account {}, "INBOX.Trash", 42, [], false, null)

    /var/www/nextcloud/lib/private/AppFramework/Http/Dispatcher.php - line 170:

    OCA\Mail\Controller\FoldersController->sync(1, "SU5CT1guVHJhc2g=", [], true, null)

    /var/www/nextcloud/lib/private/AppFramework/Http/Dispatcher.php - line 99:

    OC\AppFramework\Http\Dispatcher->executeController(OCA\Mail\Con ... {}, "sync")

    /var/www/nextcloud/lib/private/AppFramework/App.php - line 125:

    OC\AppFramework\Http\Dispatcher->dispatch(OCA\Mail\Con ... {}, "sync")

    /var/www/nextcloud/lib/private/AppFramework/Routing/RouteActionHandler.php - line 47:

    OC\AppFramework\App::main("OCA\\Mail\\ ... r", "sync", OC\AppFramew ... {}, { accountId: ... "})

    <<closure>>

    OC\AppFramework\Routing\RouteActionHandler->__invoke({ accountId: ... "})

    /var/www/nextcloud/lib/private/Route/Router.php - line 299:

    call_user_func(OC\AppFramew ... {}, { accountId: ... "})

    /var/www/nextcloud/lib/base.php - line 1008:

    OC\Route\Router->match("/apps/mail/ ... c")

    /var/www/nextcloud/index.php - line 38:

    OC::handleRequest()

### Mail app

**Mail app version:** 1.3.0

**Mailserver or service:** Hetzner

### Server configuration

**Operating system**: Debian 9

**Web server:** Apache

**Database:** MariaDB

**PHP version:** 7.3.2

**Nextcloud Version:** 18.0.3

#### Client configuration
**Browser:** Firefox 75.0

**Operating system:** Mac Catalina 10.15.3

**Comment by ChristophWurst** [comment]
Fixed by #2912 

Duplicate of #2910 #2908

**Comment by laurentiu2** [comment]
Partially solved with 1.3.1
After update I got the following error:

TypeError: Argument 7 passed to OCA\Mail\Controller\MessagesController::__construct() must be of the type string, null given

    <<closure>>

    OCA\Mail\Controller\MessagesController->__construct("mail", OC\AppFramework\Http\Request {}, OCA\Mail\Ser ... {}, OCA\Mail\Service\MailManager {}, OCA\Mail\Ser ... {}, OCA\Mail\Ser ... }}, null, null, OC\AppFramework\Logger {}, OC\L10N\LazyL10N {}, OC\Files\Type\Detection {}, OC\URLGenerator {})

    /var/www/nextcloud/lib/private/AppFramework/Utility/SimpleContainer.php - line 83:

    ReflectionClass->newInstanceArgs([ "mail",OC\ ... }])

    /var/www/nextcloud/lib/private/AppFramework/Utility/SimpleContainer.php - line 100:

    OC\AppFramework\Utility\SimpleContainer->buildClass(ReflectionCl ... "})

    /var/www/nextcloud/lib/private/AppFramework/Utility/SimpleContainer.php - line 115:

    OC\AppFramework\Utility\SimpleContainer->resolve("OCA\\Mail\\ ... r")

    /var/www/nextcloud/lib/private/AppFramework/DependencyInjection/DIContainer.php - line 415:

    OC\AppFramework\Utility\SimpleContainer->query("OCA\\Mail\\ ... r")

    /var/www/nextcloud/lib/private/AppFramework/DependencyInjection/DIContainer.php - line 386:

    OC\AppFramework\DependencyInjection\DIContainer->queryNoFallback("OCA\\Mail\\ ... r")

    /var/www/nextcloud/lib/private/AppFramework/App.php - line 112:

    OC\AppFramework\DependencyInjection\DIContainer->query("OCA\\Mail\\ ... r")

    /var/www/nextcloud/lib/private/AppFramework/Routing/RouteActionHandler.php - line 47:

    OC\AppFramework\App::main("OCA\\Mail\\ ... r", "index", OC\AppFramew ... {}, { accountId: ... "})

    <<closure>>

    OC\AppFramework\Routing\RouteActionHandler->__invoke({ accountId: ... "})

    /var/www/nextcloud/lib/private/Route/Router.php - line 299:

    call_user_func(OC\AppFramew ... {}, { accountId: ... "})

    /var/www/nextcloud/lib/base.php - line 1008:

    OC\Route\Router->match("/apps/mail/ ... s")

    /var/www/nextcloud/index.php - line 38:

    OC::handleRequest()

I have 4 mailboxes configured.
I can see messages from all folders (inbox, sent, trash) for 3 of them. For the 4th I cannot see trash - it's just indexing indefinitely there, but I have only 114 messages in Trash.

After sending a new mail I cannot see it in 'Sent'.
It does arrive at destination, so it's working..
It's kinda duplicate of  #2915 but wrote here to put the error. Hope it helps.

The only error after update, when I deleted some email from 'Sent' - it may be related to  #2487 the behaviour is the same:
{"reqId":"q987hh9JsU3tSINrS3i2","level":3,"time":"2020-04-16T13:30:42+00:00","remoteAddr":"5.15.117.204","user":"laurentiu","app":"PHP","method":"GET","url":"/index.php/apps/mail/api/accounts/1/folders/SU5CT1guU2VudA%3D%3D/messages/1334/body","message":"Undefined offset: 1333 at /var/www/nextcloud/apps/mail/lib/Cache/Cache.php#145","userAgent":"Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:75.0) Gecko/20100101 Firefox/75.0","version":"18.0.3.0","id":"5e985e3a9f391"}

--
Looks awesome 1.3! 😃😃😃

**Comment by ChristophWurst** [comment]
> TypeError: Argument 7 passed to OCA\Mail\Controller\MessagesController::__construct() must be of the type string, null given

This is weird. That should only happen when you send an unauthenticated request. Does the error persist?

> For the 4th I cannot see trash - it's just indexing indefinitely there, but I have only 114 messages in Trash.

Sounds just like https://github.com/nextcloud/mail/issues/2909. There the sync also never finishes for some reason.

> It's kinda duplicate of #2915 but wrote here to put the error. Hope it helps.

Yep, that's the same :)

> The only error after update, when I deleted some email from 'Sent' - it may be related to #2487 the behaviour is the same:

Yeah, this pops up occasionally. But it's not a hard or critical error. We should fix it, though :)

> Looks awesome 1.3! smileysmileysmiley

Thanks. I'll forward that to the team :)

**Comment by laurentiu2** [comment]
> This is weird. That should only happen when you send an unauthenticated request. Does the error persist?
No. It just appeared once.
However, if navigating from 'All Inboxes' directly to 'Sent' in any account, the list of emails is visible, but no the content.
You have to go to 'Inbox' first, and only after that to 'Sent' to display the content.
No errors in Logging.

**Comment by laurentiu2** [comment]
Ups, the ability to create new folder while saving attachments has disappeared in 1.3.1
![Screenshot 2020-04-16 at 17 02 19](https://user-images.githubusercontent.com/29857699/79465738-6b800d00-8004-11ea-8504-19f93273d607.png)

**Comment by ChristophWurst** [comment]
> Ups, the ability to create new folder while saving attachments has disappeared in 1.3.1

Whoops indeed. Please open a separate ticket.
